packages/lesswrong/components/admin/RandomUserPage.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useEffect, useState } from 'react';
+import { Components, fragmentTextForQuery, registerComponent } from '../../lib/vulcan-lib';
+import { userIsAdminOrMod } from '../../lib/vulcan-users/permissions';
+import { useCurrentUser } from '../common/withUser';
+import { gql, useLazyQuery } from '@apollo/client';
+import Button from '@material-ui/core/Button';
+
+const styles = (theme: ThemeType): JssStyles => ({
+  root: {
+    marginTop: -theme.spacing.mainLayoutPaddingTop,
+    padding: "24px 36px 60px",
+    backgroundColor: theme.palette.background.paper,
+  },
+  body: {
+    fontFamily: theme.palette.fonts.sansSerifStack,
+    marginTop: 4,
+    marginBottom: 20,
+  },
+  button: {
+    padding: "12px 24px",
+    marginRight: 16,
+    backgroundColor: theme.palette.grey[60],
+  },
+});
+
+const RandomUserPage = ({classes}: {
+  classes: ClassesType,
+}) => {
+  const currentUser = useCurrentUser();
+  const [newTabKeyHeld, setNewTabKeyHeld] = useState(false);
+  const [recievedNewResults, setRecievedNewResults] = useState(false);
+  const [getRandomUser, {loading, data}] = useLazyQuery(gql`
+    query randomUser($userIsAuthor: String!) {
+      GetRandomUser(userIsAuthor: $userIsAuthor) {
+        ...UsersMinimumInfo
+      }
+    }
+    ${fragmentTextForQuery('UsersMinimumInfo')}
+  `, {
+    onCompleted: (data) => {
+      if (!data.GetRandomUser) return;
+      // You might imagine we could redirect here, but we don't have the status
+      // of the new tab key, so we use the useEffect below
+      setRecievedNewResults(true);
+    },
+    fetchPolicy: "no-cache",
+  });
+  // Redirect to the user page
+  // useEffect combined with recievedNewResults ensures that we only redirect once
+  useEffect(() => {
+    if (recievedNewResults) {
+      const user: UsersMinimumInfo|undefined = data?.GetRandomUser;
+      if (!user) {
+        // eslint-disable-next-line no-console
+        console.error("No user found");
+        return;
+      }
+      // Redirect to the user page
+      const url = `/users/${user.slug}`;
+      if (newTabKeyHeld) {
+        window.open(url, "_blank")?.focus();
```

Comment:
This whole useEffect crap is for this line, which I couldn't figure out how to do any more elegant way.

The .focus() is intended to keep the focus on the original tab, but it doesn't work, and apparently Chrome doesn't want to allow JS to open tabs in the background so it's not possible? I decided I didn't care.

Reply:
This seems convoluted but it's not immediately obvious to me how to get this behaviour any other way, so I guess it's fine 🤷‍♂️